app/src/main/res/layout/pin_password_activity.xml

```diff
@@ -56,146 +56,153 @@
-              android:contentDescription="@{viewModel.showPassword ? @string/password_shown_icon : @string/password_hidden_icon}"
-              app:srcCompat="@{viewModel.showPassword ? @drawable/ic_show_eye_icon : @drawable/ic_hide_eye_icon}" />
+            <TextView
+              android:id="@+id/pin_password_error_text_view"
+              style="@style/TextViewStart"
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:layout_marginStart="8dp"
+              android:layout_marginTop="2dp"
+              android:layout_marginEnd="8dp"
+              android:text="@string/pin_password_incorrect_pin"
+              android:textColor="@color/edit_text_error"
+              android:textSize="16sp"
+              android:visibility="@{viewModel.showError ? View.VISIBLE : View.INVISIBLE}"
+              app:layout_constraintEnd_toEndOf="parent"
+              app:layout_constraintHorizontal_bias="0.0"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toBottomOf="@+id/enter_pin_text_view" />
 
             <TextView
-              android:id="@+id/show_hide_password_text_view"
-              style="@style/TextViewCenter"
-              android:layout_width="wrap_content"
-              android:layout_height="match_parent"
-              android:layout_gravity="center_horizontal"
-              android:layout_marginTop="4dp"
-              android:fontFamily="sans-serif-medium"
-              android:text="@{viewModel.showPassword ? @string/pin_password_hide : @string/pin_password_show}"
+              android:id="@+id/forgot_pin"
+              style="@style/Heading5ViewStart"
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:layout_marginTop="24dp"
+              android:minHeight="48dp"
+              android:paddingTop="8dp"
+              android:text="@string/pin_password_forgot_pin"
               android:textAllCaps="true"
-              android:textColor="@color/oppia_primary_text"
-              android:textSize="12sp" />
-          </LinearLayout>
+              android:textColor="@color/oppia_primary"
+              app:layout_constraintEnd_toEndOf="parent"
+              app:layout_constraintHorizontal_bias="0.0"
+              app:layout_constraintStart_toStartOf="parent"
+              app:layout_constraintTop_toBottomOf="@+id/pin_password_error_text_view" />
+          </androidx.constraintlayout.widget.ConstraintLayout>
         </androidx.constraintlayout.widget.ConstraintLayout>
-      </androidx.constraintlayout.widget.ConstraintLayout>
       </ScrollView>
       <View
         android:layout_width="match_parent"
         android:layout_height="6dp"
         android:background="@drawable/toolbar_drop_shadow" />
     </FrameLayout>
   </androidx.constraintlayout.widget.ConstraintLayout>
-</layout>
+</layout>
```

Comment:
EOF is missing, add one blank line at the end.

Reply:
done